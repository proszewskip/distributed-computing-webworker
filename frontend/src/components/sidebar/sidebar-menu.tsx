import { Button, Pane } from 'evergreen-ui';
import Link, { LinkProps } from 'next/link';
import React, { StatelessComponent } from 'react';

export interface MenuItem {
  title: string;
  href: LinkProps['href'];
  isActive: () => boolean;
}

export interface SidebarMenuProps {
  items: MenuItem[];
}

export const SidebarMenu: StatelessComponent<SidebarMenuProps> = (props) => {
  return (
    <Pane display="flex" flexDirection="column" alignItems="stretch">
      {props.items.map((item) => (
        <Link href={item.href} passHref={true} key={item.title}>
          <Button is="a" appearance="minimal" isActive={item.isActive()}>
            {item.title}
          </Button>
        </Link>
      ))}
    </Pane>
  );
};
