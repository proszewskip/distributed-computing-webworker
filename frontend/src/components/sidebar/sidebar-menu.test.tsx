import React from 'react';
import renderer from 'react-test-renderer';

import { SidebarMenu, SidebarMenuProps } from './sidebar-menu';

describe('SidebarMenu', () => {
  it('should match snapshot', () => {
    const menuItems: SidebarMenuProps['items'] = [
      {
        route: 'abc',
        title: 'Abc',
        isActive: () => true,
      },
      {
        route: 'notactive',
        title: 'Not active',
        isActive: () => false,
      },
    ];

    const instance = renderer.create(<SidebarMenu items={menuItems} />);
    expect(instance.toJSON()).toMatchSnapshot();
  });
});
