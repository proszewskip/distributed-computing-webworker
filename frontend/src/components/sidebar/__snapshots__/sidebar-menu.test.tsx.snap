// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SidebarMenu should match snapshot 1`] = `
<div
  className="ðŸ“¦dspl_flex ðŸ“¦flx-drct_column ðŸ“¦algn-itms_stretch ðŸ“¦box-szg_border-box css-nil"
>
  <a
    className="css-bsyznc ðŸ“¦fnt-fam_b77syt ðŸ“¦mt_0px ðŸ“¦fnt-sze_12px ðŸ“¦f-wght_500 ðŸ“¦ln-ht_32px ðŸ“¦ltr-spc_0 ðŸ“¦btrr_3px ðŸ“¦bbrr_3px ðŸ“¦btlr_3px ðŸ“¦bblr_3px ðŸ“¦pt_0px ðŸ“¦pb_0px ðŸ“¦pr_16px ðŸ“¦pl_16px ðŸ“¦ml_0px ðŸ“¦mr_0px ðŸ“¦mb_0px ðŸ“¦h_32px ðŸ“¦pst_relative ðŸ“¦dspl_inline-flex ðŸ“¦algn-itms_center ðŸ“¦flx-wrap_nowrap ðŸ“¦box-szg_border-box"
    data-active={true}
    href="abc"
    onClick={[Function]}
  >
    Abc
  </a>
  <a
    className="css-bsyznc ðŸ“¦fnt-fam_b77syt ðŸ“¦mt_0px ðŸ“¦fnt-sze_12px ðŸ“¦f-wght_500 ðŸ“¦ln-ht_32px ðŸ“¦ltr-spc_0 ðŸ“¦btrr_3px ðŸ“¦bbrr_3px ðŸ“¦btlr_3px ðŸ“¦bblr_3px ðŸ“¦pt_0px ðŸ“¦pb_0px ðŸ“¦pr_16px ðŸ“¦pl_16px ðŸ“¦ml_0px ðŸ“¦mr_0px ðŸ“¦mb_0px ðŸ“¦h_32px ðŸ“¦pst_relative ðŸ“¦dspl_inline-flex ðŸ“¦algn-itms_center ðŸ“¦flx-wrap_nowrap ðŸ“¦box-szg_border-box"
    href="notactive"
    onClick={[Function]}
  >
    Not active
  </a>
</div>
`;
